File: src/p1006/golang/solution.go
package main

func clumsy(n int) int {
	result, sign := 0, 1
	for n > 0 {
		temp := n
		n--
		if n > 0 {
			temp *= n
			n--
		}
		if n > 0 {
			temp /= n
			n--
		}
		temp *= sign
		if n > 0 {
			temp += n
			n--
		}
		result += temp
		sign = -1
	}
	return result
}

File: src/p1006/java/solution.java
class Solution {
    public int clumsy(int n) {
        int result = 0;
        for (int sign = 1; n > 0;) {
            int temp = n--;
            if (n > 0) temp *= n--;
            if (n > 0) temp /= n--;
            temp *= sign;
            if (n > 0) temp += n--;
            result += temp;
            sign = -1;
        }
        return result;
    }
}
